### ISSUE TITLE
runtimeChunk and splitChunk settings collide

### ISSUE THREAD

**Issue by timse** [webpack/webpack#7382]
# Bug report

`runtimeChunk: single` produces multiple runtime chunks if spitChunks minSize settings is very small.

**What is the current behavior?**
settings 
```
splitChunks: {
    minSize: 0
}
```

will basically ignore the `runtimeChunk: 'single'` setting and produce a runtime chunk per entry.

**If the current behavior is a bug, please provide the steps to reproduce.**

see above

**What is the expected behavior?**

a single runtime chunk for all entries.

**Other relevant information:**
webpack version:  latest
Node.js version:  latest
Operating System:  osx
Additional tools: - 

basically, this behavior can be "fixed" by removing the module count check here: https://github.com/webpack/webpack/blob/master/lib/optimize/RuntimeChunkPlugin.js#L22

which causes the unwanted behavior. However, I do not know if there is a valid reason for that checks existance.

**Comment by chrisdarroch** [comment]
I'm interested in this problem because of how having multiple runtimes plays havoc with code that expects to only ever load and be evaluated once. 

Since the webpack runtime includes a locally-scoped `installedModules` object, any double-up of the runtime means a potential for a shared module to evaluate twice. This is problematic for modules that register side-effects, like DOM event handlers.

Yeah, I know that in an ideal world, shared modules shouldn't exhibit such behaviour and should be idempotent. In reality, we've been able to register these as AMD modules and avoid multiple inclusions or evaluations of them. Only now when running the modules through webpack 4 are we experiencing this problem.

**Comment by sokra** [comment]
Could you check if this is fixed in 4.9.1?

**Comment by chrisdarroch** [comment]
Yep @sokra, I'm getting a single runtime in 4.9.1. Thanks for fixing this so quickly!
